src/main/java/uk/gov/hmcts/ccd/domain/types/ValidationContext.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.hmcts.ccd.domain.types;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.std.CaseDataContent;
+
+import java.util.Map;
+
+public class ValidationContext {
+
+    private final CaseDataContent currentCaseDataContent;
+    private final String caseTypeId;
+    private final CaseTypeDefinition caseTypeDefinition;
+    private final Map<String, JsonNode> data;
+    private String fieldId;
+    private String path;
+    private JsonNode dataValue;
+    private CaseFieldDefinition caseFieldDefinition;
+
+    public ValidationContext() {
+        this(null, null, null, null);
+    }
+
+    public ValidationContext(Map<String, JsonNode> data, CaseTypeDefinition caseTypeDefinition) {
+        this(null, null, caseTypeDefinition, data);
+    }
+
+    public ValidationContext(CaseDataContent currentCaseDataContent, String caseTypeId, CaseTypeDefinition caseTypeDefinition, Map<String, JsonNode> data) {
```

Comment:
we don't need 
`Map<String, JsonNode> data` it's already inside the first parameter `CaseDataContent currentCaseDataContent`